src/apps/react.js

```diff
@@ -114,10 +114,15 @@ export function useAppsSections(
     .sort(({ name: _name }, { name }) =>
       installed.indexOf(_name) > installed.indexOf(name) ? -1 : 0
     )
-    .sort(
-      ({ name: _name }, { name }) =>
-        installQueue.indexOf(_name) > installQueue.indexOf(name) ? -1 : 0 // place install queue on top of list
-    );
+    .sort(({ name: _name }, { name }) => {
+      // place install queue on top of list
+      // with the app being installed at the top
+      let pos1 = installQueue.indexOf(_name);
+      let pos2 = installQueue.indexOf(name);
+      pos1 = pos1 < 0 ? Number.MAX_VALUE : pos1;
+      pos2 = pos2 < 0 ? Number.MAX_VALUE : pos2;
+      return pos1 - pos2;
+    });
```

Comment:
is there a way to rework this sort to avoid the "double sort()" chain? I feel this part is not really well written today / possibly have perf issue on the complexity.

asking this now because maybe you have a hint on this part as you digged it.

Reply:
> is there a way to rework this sort to avoid the "double sort()" chain? I feel this part is not really well written today / possibly have perf issue on the complexity.

I removed the first .sort which I think was not needed 👍. (`installed` seems to be an object[] but was assumed to be an array of strings in the comparison function).